dali/pipeline/pipeline.cc

```diff
@@ -235,6 +236,43 @@ void Pipeline::AddOperator(OpSpec spec, const std::string& inst_name) {
   this->op_specs_to_serialize_.push_back(true);
 }
 
+inline int GetMemoryHint(OpSpec &spec, int index) {
+  std::vector<int> hints;
+  GetSingleOrRepeatedArg(spec, &hints, "bytes_per_sample_hint", spec.NumOutput());
+
+  if (index < (int)hints.size())
+    return hints[index];
+  else
+    return 0;
+}
+
+inline void SetMemoryHint(OpSpec &spec, int index, int value) {
+  std::vector<int> hints;
+  int no = spec.NumOutput();
+
+  DALI_ENFORCE(index < no, "Output index out of range: " +
+    std::to_string(index) + " >= " + std::to_string(no));
+
+  GetSingleOrRepeatedArg(spec, &hints, "bytes_per_sample_hint", no);
+  hints.resize(no, 0);
+  hints[index] = value;
+  spec.SetArg("bytes_per_sample_hint", hints);
+}
+
+void Pipeline::PropagateMemoryHint(OpNode &node) {
```

Comment:
It this only for MakeContiguous?

Reply:
Currently yes; I don't think we insert any other implicit nodes.